include/oneapi/tbb/detail/_concurrent_queue_base.h

```diff
@@ -329,12 +332,21 @@ class micro_queue {
         construct_item( &dst[dindex], static_cast<const void*>(&src_item) );
     }
 
-    void assign_and_destroy_item( void* dst, padded_page& src, size_type index ) {
+    // Assign the item for regular pop operation
+    void select_pop_operation( void* dst, padded_page& src, size_type index, /*do_assign = */std::true_type ) {
+        __TBB_ASSERT(dst, "Incorrect destination for assign operation during pop");
         auto& from = src[index];
         destroyer d(from);
         *static_cast<T*>(dst) = std::move(from);
     }
 
+    // No assign for pop while clearing the queue
+    void select_pop_operation( void* dst, padded_page& src, size_type index, /*do_assign = */std::false_type ) {
```

Comment:
```suggestion
    void select_pop_operation( void*, padded_page& src, size_type index, /*do_assign = */std::false_type ) {
```

Reply:
Irrelevant for the newest version of the patch